sqlacodegen/codegen.py

```diff
@@ -158,8 +158,22 @@ def add_imports(self, collector):
             if len(index.columns) > 1:
                 collector.add_import(index)
 
+    @staticmethod
+    def _convert_to_valid_identifier(name):
+        assert name, 'Identifier cannot be empty'
+        if name[0].isdigit() or iskeyword(name):
+            name = '_' + name
+        elif name == 'metadata':
+            name = 'metadata_'
+
+        return _re_invalid_identifier.sub('_', name)
+
 
 class ModelTable(Model):
+    def __init__(self, table):
+        super(ModelTable, self).__init__(table)
+        self.name = self._convert_to_valid_identifier(table.name)
```

Comment:
**_convert_to_valid_identifier** used in ModelTable

Reply:
Ah, my bad – the diffs can sometimes be confusing and my memory of the code layout is less than perfect :)